webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import Select from '../../../../../common/forms/Select';
+import Checkbox from '../../../../../common/forms/Checkbox';
+import TextInput from '../../../../../common/forms/TextInput';
+import './disk.scss';
+
+const Disk = (
+  {
+    removeDisk,
+    updateDisk,
+    name,
+    config: {
+      datastores,
+      storagePods,
+      diskModeTypes
+    },
+    storagePod,
+    dataStore,
+    size,
+    thinProvision,
+    eagerZero,
+    mode
+  }
+) => {
+  return (
+    <div className="disk-container">
+      <div className="form-group">
+        <label className="col-md-2 control-label">
+          {__('Disk name')}
+        </label>
+        <div className="col-md-4">
+          {name}
+        </div>
+      </div>
+      {!dataStore &&
+        <Select
+          label={__('Storage Pod')}
+          value={storagePod}
+          onChange={updateDisk.bind(this, 'storagePod')}
+          options={storagePods}
+        />}
+      {!storagePod &&
+        <Select
+          label={__('Data store')}
+          value={dataStore}
+          onChange={updateDisk.bind(this, 'dataStore')}
+          options={datastores}
+        />}
+
+      <Select
+        label={__('mode')}
+        value={mode}
+        onChange={updateDisk.bind(this, 'mode')}
+        options={diskModeTypes}
+      />
+
+      <TextInput
+        value={size}
+        className="text-vmware-size"
+        onChange={updateDisk.bind(this, 'size')}
```

Comment:
@matanwerbner : Sure. I think the main issue is that ruby and js have different naming conventions.
Foreman relies on fog-vsphere to create a VM with volumes. fog-vsphere specifies a volume model that currently has the attributes `size` and `size_gb`. They are backed by the same property.
https://github.com/fog/fog-vsphere/blob/d213ec1562d512f3d8913106d7af4ae1b93bd218/lib/fog/vsphere/models/compute/volume.rb#L26-L33

You can easily use `size` instead of `size_gb` but a user should only see the size in GB. We could also replace this via a more intelligent text field with a spinner.